org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/DisplayProjectRuleDescriptionJob.java

```diff
@@ -19,72 +19,47 @@
  */
 package org.sonarlint.eclipse.ui.internal.job;
 
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.UUID;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
-import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.swt.widgets.Display;
 import org.sonarlint.eclipse.core.SonarLintLogger;
 import org.sonarlint.eclipse.core.internal.backend.SonarLintBackendService;
 import org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob;
 import org.sonarlint.eclipse.core.resource.ISonarLintProject;
 import org.sonarlint.eclipse.ui.internal.rule.RuleDetailsPanel;
-import org.sonarsource.sonarlint.core.rpc.protocol.backend.rules.EffectiveRuleDetailsDto;
-import org.sonarsource.sonarlint.core.rpc.protocol.backend.rules.ImpactDto;
-import org.sonarsource.sonarlint.core.rpc.protocol.common.ImpactSeverity;
-import org.sonarsource.sonarlint.core.rpc.protocol.common.IssueSeverity;
-import org.sonarsource.sonarlint.core.rpc.protocol.common.RuleType;
-import org.sonarsource.sonarlint.core.rpc.protocol.common.SoftwareQuality;
 
 /** Update "web browser" view for the project rule description (maybe context based on connection) in a separate thread */
 public class DisplayProjectRuleDescriptionJob extends AbstractSonarProjectJob {
-  private final String ruleKey;
-  private final RuleType issueType;
-  private final IssueSeverity issueSeverity;
-  private final Map<SoftwareQuality, ImpactSeverity> issueImpacts;
-  private final String contextKey;
+  private final UUID issueId;
   private final RuleDetailsPanel ruleDetailsPanel;
 
-  public DisplayProjectRuleDescriptionJob(ISonarLintProject project, String ruleKey, @Nullable RuleType issueType,
-    @Nullable IssueSeverity issueSeverity, Map<SoftwareQuality, ImpactSeverity> issueImpacts,
-    @Nullable String contextKey, RuleDetailsPanel ruleDetailsPanel) {
-    super("Fetching rule description for rule '" + ruleKey + "'...", project);
-    this.ruleKey = ruleKey;
-    this.issueType = issueType;
-    this.issueSeverity = issueSeverity;
-    this.issueImpacts = issueImpacts;
-    this.contextKey = contextKey;
+  public DisplayProjectRuleDescriptionJob(ISonarLintProject project, UUID issueId,
+    RuleDetailsPanel ruleDetailsPanel) {
+    super("Fetching rule description for issue '" + issueId + "'...", project);
```

Comment:
Is it visible in the UI? If yes, the issue ID is purely technical and does not really make sense for the user. I'd rather show the rule key even though it's in the context of a specific issue

Reply:
No, it is the title of the job. So if it fails this can be debugged as the job will be shown as failed with it's title and information ^^